src/ConformalTracking.cc

```diff
@@ -1386,6 +1386,18 @@ void ConformalTracking::extendTracks(UniqueKDTracks& conformalTracks, SharedKDCl
       if (deltaChi2 > chi2cut || deltaChi2zs > chi2cut)
         continue;
 
+      bool onSameLayer = false;
+
+      for (unsigned int hitOnTrack = 0; hitOnTrack < conformalTracks[currentTrack]->m_clusters.size(); hitOnTrack++) {
+        SKDCluster clusterOnTrack = conformalTracks[currentTrack]->m_clusters.at(hitOnTrack);
```

Comment:
`SKDCluster const& clusterOnTrack = ....` (or with `auto const&`)
Otherwise the shared_ptr is copied